pkg/backends/otlp/internal/data/histograms.go

```diff
@@ -0,0 +1,91 @@
+package data
+
+import (
+	"math"
+	"slices"
+
+	v1metrics "go.opentelemetry.io/proto/otlp/metrics/v1"
+	"golang.org/x/exp/maps"
+)
+
+type Histogram struct {
+	raw *v1metrics.Histogram
+}
+
+type HistogramDataPoint struct {
+	raw *v1metrics.HistogramDataPoint
+}
+
+func NewHistogram(datapoints ...HistogramDataPoint) Histogram {
+	ht := Histogram{
+		raw: &v1metrics.Histogram{
+			AggregationTemporality: v1metrics.AggregationTemporality_AGGREGATION_TEMPORALITY_DELTA,
+			DataPoints:             make([]*v1metrics.HistogramDataPoint, 0, len(datapoints)),
+		},
+	}
+
+	for i := 0; i < len(datapoints); i++ {
+		ht.raw.DataPoints = append(
+			ht.raw.DataPoints,
+			datapoints[i].raw,
+		)
+	}
+
+	return ht
+}
+
+func WithHistogramDataPointTimeStamp(ts int64) func(HistogramDataPoint) {
```

Comment:
Are timestamps optional for metrics? If not then we should force it in the constructor